mlpf/tfmodel/model_setup.py

```diff
@@ -192,15 +198,20 @@ def prepare_callbacks(model, outdir, X_val, y_val, dataset_transform, num_output
     terminate_cb = tf.keras.callbacks.TerminateOnNaN()
     callbacks += [terminate_cb]
 
+    cp_dir = Path(outdir) / "weights"
+    cp_dir.mkdir(parents=True, exist_ok=True)
     cp_callback = tf.keras.callbacks.ModelCheckpoint(
-        filepath=outdir + "/weights-{epoch:02d}-{val_loss:.6f}.hdf5",
+        filepath=str(cp_dir / "weights-{epoch:02d}-{val_loss:.6f}.hdf5"),
         save_weights_only=True,
         verbose=0
     )
     cp_callback.set_model(model)
     callbacks += [cp_callback]
 
-    cb = CustomCallback(outdir, X_val, y_val, dataset_transform, num_output_classes)
+    history_path = Path(outdir) / "history"
+    history_path.mkdir()
```

Comment:
to support retraining an existing model with model_setup (for backwards compatibility), we need to add a check here in case the directory already exists:

```
$ singularity exec --nv ~/HEP-KBFI/singularity/base.simg python3 mlpf/launcher.py --action train --model-spec parameters/cms-gnn-dense.yaml --ntrain 1000 --ntest 1000 --weights experiments/cms-gnn-dense-53a457a3.joosep-desktop/weights/weights-02-28.025951.hdf5 
...
Traceback (most recent call last):
  File "mlpf/launcher.py", line 55, in <module>
    tfmodel.model_setup.main(args, yaml_path, config)
  File "/home/joosep/particleflow/mlpf/tfmodel/model_setup.py", line 711, in main
    callbacks = prepare_callbacks(
  File "/home/joosep/particleflow/mlpf/tfmodel/model_setup.py", line 212, in prepare_callbacks
    history_path.mkdir()
  File "/usr/lib/python3.8/pathlib.py", line 1284, in mkdir
    self._accessor.mkdir(self, mode)
FileExistsError: [Errno 17] File exists: 'experiments/cms-gnn-dense-53a457a3.joosep-desktop/history'
```